### ISSUE TITLE
Deleting a redundant work queue breaks link in the UI

### ISSUE THREAD

**Issue by Bada-S** [PrefectHQ/prefect#8494]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

If you create a work queue with the same name in two different work pools- for example a test work queue in the default work pool and a test work queue in a prod work pool - and then delete one of those work queues, it breaks the link for the work queue shown in the UI. 

This issue was originally flagged by Block and I reproduced it by building the two deployments  shown below and then deleting only the work queue in the default work pool

<img width="1111" alt="Screenshot 2023-02-10 at 4 34 08 PM" src="https://user-images.githubusercontent.com/69161193/218213819-5c227b6c-d8cd-4253-949f-f032d4032bf5.png">

<img width="1108" alt="Screenshot 2023-02-10 at 4 32 26 PM" src="https://user-images.githubusercontent.com/69161193/218213500-6fa8e8d7-2e90-42c3-bea1-596b5638bcff.png">

### Reproduction

```python3
prefect deployment build flow.py:hi -n test -q test --pool test-pool
prefect deployment build flow.py:hi -n test2 -q test --pool default-agent-pool
```

### Versions

```Text
Prefect 2.8.0
```

**Comment by serinamarie** [comment]
Thanks for the issue! @marichka-offen would you be able to look into this?

**Comment by zhen0** [comment]
This may be as expected.  @marichka-offen will look into it.

**Comment by marichka-offen** [comment]
@Bada-S I am unable to reproduce the issue. If you delete a work queue from the pool assigned to a specific deployment, that deployment loses the link to that work pool and that work queue becomes disabled. What you can do is go to Edit deployment and select a pool and a new existing queue. It does not however impact a queue with the same name in a different pool. As an idea, since we expect all queues to have pools now, we can hide disabled queues completely if there's no pool it tied to

**Comment by Bada-S** [comment]
@marichka-offen When I tested out deleting one of the work queues, it didn't just break the link to the work queue that got deleted(that is to be expected), it broke the link to the other work queue as well. I can see that the deployment associated with the work queue that was not deleted has a broken link to the work queue. I will try again and see if this occurs a second, because it is definitely possible that there was user error on both of our accounts.

**Comment by Bada-S** [comment]
I did see that editing the deployment does indeed restore the link so I can suggest that as a fix for now

**Comment by Bada-S** [comment]
So I was able to reproduce the error a second time. My set up was to use two deployments- "test" and "test2." "test" would be deployed with "test" work queue in the "test-pool" work pool and "test2" would be deployed with "test" work queue in the "default-agent-pool" work pool. I noticed on my second time testing the issue, that the link for both deployments linked to the "default-agent-pool" irrespective of the deployment specifying the "test-pool" work pool. I think that would explain why the links are breaking in both deployments. There must be some logic that dictates that the first instance of a work queue name is the one that is linked to in the ui
![image](https://user-images.githubusercontent.com/69161193/218577260-421bfc0d-3bde-4327-bde8-1f58cba3bbfb.png)
